scarlet/morphology.py

```diff
@@ -46,6 +48,24 @@ def __init__(self, frame, *parameters, bbox=None):
 
         super().__init__(*parameters)
 
+    def shrink_box(self, image, thresh=0):
```

Comment:
Feels like this should belong in the Box class, just after the `grow` method.

Reply:
not quite. `shrink_box` depends on the image (to find out by how much it can shrink), whereas `box.grow` just takes a radius.

That brings up a concern: `box.grow` does not need to respect the fixed set of box sizes we allow to keep the cache small. @fred3m have you checked, or is that not a concern?

Reply:
This is a different method than growing the `Box`. `Box.grow` takes a `radius` and the method is used to increase the size of the box by that amount in each direction (top, bottom, left, right) but keep the same center. It is not automatically fit, so it's up to the user to restrict the size if they wish to do so (in my case this is not an issue because I only grow the box to create a new image that contains the source convolved with the PSF).

Since this is fitting the box size, and the algorithm to do that will be dependent on the use case, I think that it's more appropriate to keep it outside of the Box` class.

Reply:
I hesitated making the distinction the first time: I don't suggest to put the `shrink_box` method in `Box`, but to have a `shrink` method in `Box` that we would call here, and that we could call anytime we need boxes to shrink.

Reply:
Though it can be done in a separate PR. No need to hold this one here, but I feel like it corresponds more to the spirit of how boxes work and also could possibly be useful for other morphologies.

Reply:
that's why it's implemented at the `Morphology` level, not in `ImageMorphology`